include/tvm/runtime/memory.h

```diff
@@ -73,6 +74,26 @@ class ObjAllocatorBase {
     ptr->deleter_ = Handler::Deleter();
     return ObjectPtr<T>(ptr);
   }
+
+  /*!
+   * \tparam ArrayType The type to be allocated.
+   * \tparam ElemType The type of array element.
+   * \tparam Args The constructor signature.
+   * \param num_elems The number of array elements.
+   * \param args The arguments.
+   */
+  template<typename ArrayType, typename ElemType, typename... Args>
+  inline ObjectPtr<ArrayType> make_inplace_array(size_t num_elems, Args&&... args) {
+    using Handler = typename Derived::template Handler<ArrayType, ElemType>;
+    static_assert(std::is_base_of<Object, ArrayType>::value,
+                  "make_node can only be used to create NodeBase");
```

Comment:
Hmmm i am not sure if the error message is proper

Reply:
Good catch! fixed.